Tests/js/views/assets/ez-subitemlistview-tests.js

```diff
@@ -530,10 +546,367 @@ YUI.add('ez-subitemlistview-tests', function (Y) {
+                    });
+                    that.wait();
+                });
+            });
+            this.wait();
+        },
+
+        "Should display an error icon if priority input is not correctly set": function () {
+            var c = this.view.get('container'),
+                that = this,
+                selectedCell,
+                fieldInput;
+
+            this.view.render();
+
+            fieldInput =  c.one('.ez-subitem-priority-input');
+            selectedCell = c.one('#priority-cell-' + this.locationId);
+            fieldInput.simulateGesture('tap');
+
+            this.waitFor(function () {
+                return selectedCell.hasClass("ez-subitem-selected-priority-cell");
+            }, function () {
+                    fieldInput.set('value', 'BAD PRIORITY');
+                    fieldInput.simulate('blur');
+                    that.waitFor(function () {
+                        return selectedCell.hasClass("ez-subitem-error-priority-cell");
+                    }, function () {
+                        Y.Assert.isFalse(selectedCell.hasClass("ez-subitem-selected-priority-cell"), "Validate and Cancel buttons should be hidden");
+                        Y.Assert.isTrue(selectedCell.hasClass("ez-subitem-error-priority-cell"), "Error icon should be displayed");
+                    },2000);
+                },2000
+            );
+        },
+
+        "Should hide the error icon if priority input is correctly set again": function () {
+            var c = this.view.get('container'),
+                that = this,
+                selectedCell,
+                fieldInput;
+
+            this.view.render();
+
+            fieldInput =  c.one('.ez-subitem-priority-input');
+            selectedCell = c.one('#priority-cell-' + this.locationId);
+            fieldInput.simulateGesture('tap');
+
+            this.waitFor(function () {
+                    return selectedCell.hasClass("ez-subitem-selected-priority-cell");
+                }, function () {
+                    fieldInput.set('value', 'BAD PRIORITY');
+                    fieldInput.simulate('blur');
+                    that.waitFor(function () {
+                        return selectedCell.hasClass("ez-subitem-error-priority-cell");
+                    }, function () {
+                        fieldInput.set('value', '42');
+                        fieldInput.simulate('blur');
+                        that.waitFor(function () {
+                            return selectedCell.hasClass("ez-subitem-selected-priority-cell");
+                            }, function () {
+                                Y.Assert.isFalse(selectedCell.hasClass("ez-subitem-error-priority-cell"), "Error icon should be hidden");
+                        }, 2000);
+                    },2000);
+                },2000
+            );
+        },
+
+        "Should fire updatePriority and hide priority buttons on tap on the validate button": function () {
```

Comment:
actually that should not be the case anymore, so I wonder why this test passes ?